tools/parquetDDL/main.cpp

```diff
@@ -0,0 +1,285 @@
+#include <iostream>
+#include <string>
+#include <arrow/api.h>
+#include <arrow/io/api.h>
+#include <parquet/exception.h>
+#include <parquet/arrow/reader.h>
+#include <vector>
+#include <fstream>
+#include <unistd.h>
+
+enum STATUS_CODE
+{
+  NO_ERROR,
+  EMPTY_FIELD,
+  UNSUPPORTED_DATA_TYPE,
+  UNSUPPORTED_FILE_TYPE,
+  FILE_NUM_ERROR
+};
+
+/**
+ * print the usage information
+*/
+static void usage()
+{
+  std::cout << "usage: " << std::endl;
+  std::cout << "Reading parquet then output its corresponding .ddl file." << std::endl;
+  std::cout << "mcs_parquet_ddl [input_parquet_file] [output_ddl_file]" << std::endl;
+}
+
+/**
+ * get the schema of the parquet file
+*/
+void getSchema(std::string filePath, std::shared_ptr<arrow::Schema>* parquetSchema)
+{
+  std::shared_ptr<arrow::io::ReadableFile> infile;
+  PARQUET_ASSIGN_OR_THROW(infile, arrow::io::ReadableFile::Open(filePath, arrow::default_memory_pool()));
```

Comment:
There is no corresponding catch for these throws so the utility will crash with SIGABRT, that is not good UX IMHO.